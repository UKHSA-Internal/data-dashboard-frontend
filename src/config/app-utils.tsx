/**
 * Util function to get the revalidate value for NextJs getStaticProps
 * Passing 'false' will disable revalidation altogether and return
 * a stale version of the initial built page to all users.
 */
export const getStaticPropsRevalidateValue = () => {
  if (process.env.NEXT_REVALIDATE_TIME === 'false') {
    return false
  }

  return Number(process.env.NEXT_REVALIDATE_TIME)
}
