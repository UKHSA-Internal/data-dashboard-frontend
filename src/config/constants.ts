/**
 * The NextJS Api Route path for chart download
 */
export const chartExportApiRoutePath = '/api/download/chart'

/**
 * The NextJS Api Route path for bulk download
 */
export const bulkExportApiRoutePath = '/api/download/bulk'

/**
 * The default file format in which bulk downloaded charts are downloaded as
 */
export const bulkExportFormat: 'csv' | 'json' = 'csv'

/**
 * The maximum columns per table in the tabular data view
 */
export const chartTableMaxColumns = {
  narrow: 2,
  wide: 3,
} as const

/**
 * Chart size variants
 */
export const chartSizes = {
  narrow: {
    width: 515,
    height: 260,
  },
  wide: {
    width: 1100,
    height: 260,
  },
} as const

/**
 * The default file format in which charts are displayed as
 */
export const chartFormat = 'svg'
