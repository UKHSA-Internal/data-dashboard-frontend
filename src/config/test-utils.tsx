/* eslint-disable import/export,no-restricted-imports */
import React, { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { I18nextProvider } from 'react-i18next'

import i18n from '../../config/i18n/i18nForTests'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
