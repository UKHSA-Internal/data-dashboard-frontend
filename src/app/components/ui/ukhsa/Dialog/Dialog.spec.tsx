import userEvent from '@testing-library/user-event'
import Link from 'next/link'

import { render, screen } from '@/config/test-utils'

import {
  Dialog,
  DialogContent,
  DialogDescription,
  DialogFooter,
  DialogHeader,
  DialogOverlay,
  DialogTitle,
  DialogTrigger,
} from './Dialog'

const dialogHeader = (
  <DialogHeader>
    <DialogTitle>Dialog title</DialogTitle>
    <DialogDescription>Dialog description</DialogDescription>
  </DialogHeader>
)

const dialogContent = <DialogContent>{dialogHeader}</DialogContent>

describe('Dialog', () => {
  beforeEach(() => {
    console.error = jest.fn()
  })

  test('Opening a dialog from a trigger', async () => {
    render(
      <Dialog>
        <DialogTrigger>Open</DialogTrigger>
        {dialogContent}
      </Dialog>
    )

    const triggerElement = screen.getByRole('button', { name: 'Open' })
    expect(triggerElement).toHaveAttribute('aria-expanded', 'false')
    expect(triggerElement).toHaveAttribute('aria-haspopup', 'dialog')
    expect(triggerElement).toHaveAttribute('aria-controls', 'radix-:r0:')

    expect(screen.queryByRole('dialog', { name: 'Dialog title' })).not.toBeInTheDocument()

    await userEvent.click(triggerElement)

    expect(triggerElement).toHaveAttribute('aria-expanded', 'true')

    const dialog = screen.getByRole('dialog', { name: 'Dialog title' })

    expect(dialog).toBeInTheDocument()
    expect(screen.getByText('Dialog description')).toBeInTheDocument()
  })

  test('Defaulting open on page load', async () => {
    render(<Dialog defaultOpen>{dialogContent}</Dialog>)

    expect(screen.getByRole('dialog', { name: 'Dialog title' })).toBeInTheDocument()
    expect(screen.getByText('Dialog description')).toBeInTheDocument()
  })

  test('Closing a dialog', async () => {
    render(<Dialog defaultOpen>{dialogContent}</Dialog>)

    expect(screen.getByRole('dialog', { name: 'Dialog title' })).toBeInTheDocument()
    await userEvent.click(screen.getByRole('button', { name: 'Close' }))
    expect(screen.queryByRole('dialog', { name: 'Dialog title' })).not.toBeInTheDocument()
  })

  test('Closing a dialog with a custom close button', async () => {
    render(
      <Dialog defaultOpen>
        <DialogContent closeButton={<Link href="/custom-href">Custom close</Link>}>{dialogHeader}</DialogContent>
      </Dialog>
    )

    expect(screen.getByRole('dialog', { name: 'Dialog title' })).toBeInTheDocument()

    const customCloseBtn = screen.getByRole('link', { name: 'Custom close' })
    expect(customCloseBtn).toBeInTheDocument()
    await userEvent.click(customCloseBtn)

    expect(screen.queryByRole('dialog', { name: 'Dialog title' })).not.toBeInTheDocument()
  })

  test('Fullscreen', () => {
    render(
      <Dialog defaultOpen>
        <DialogContent fullscreen>{dialogHeader}</DialogContent>
      </Dialog>
    )

    expect(screen.getByRole('dialog', { name: 'Dialog title' })).toHaveClass('w-full h-full')
  })

  test('Background overlay', () => {
    const { container } = render(
      <Dialog defaultOpen>
        <DialogOverlay>Nothing to see here</DialogOverlay>
      </Dialog>
    )

    expect(container.firstChild).toHaveClass(
      'fixed inset-0 z-50 bg-[black]/70  data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0'
    )
  })

  test('Custom footer', () => {
    render(
      <Dialog defaultOpen>
        <DialogContent fullscreen>
          {dialogHeader}
          <DialogFooter>Dialog footer</DialogFooter>
        </DialogContent>
      </Dialog>
    )

    expect(screen.getByText('Dialog footer')).toBeInTheDocument()
  })
})
