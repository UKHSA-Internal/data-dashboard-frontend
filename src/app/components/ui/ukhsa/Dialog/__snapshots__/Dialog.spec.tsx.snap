// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Opening a dialog from a trigger 1`] = `
<div>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    Open
  </button>
</div>
`;

exports[`Dialog Opening a dialog from a trigger 2`] = `
<div
  aria-describedby="radix-:r2:"
  aria-labelledby="radix-:r1:"
  class="fixed left-1/2 top-1/2 z-50 grid w-full -translate-x-1/2 -translate-y-1/2 gap-4 border-0 bg-white p-6 pt-[5rem] shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] max-w-lg"
  data-state="open"
  id="radix-:r0:"
  role="dialog"
  style="pointer-events: auto;"
  tabindex="-1"
>
  <div
    class="flex flex-col text-center sm:text-left"
  >
    <h2
      class="govuk-heading-l"
      id="radix-:r1:"
    >
      Dialog title
    </h2>
    <p
      class="govuk-body"
      id="radix-:r2:"
    >
      Dialog description
    </p>
  </div>
  <button
    class="focus:govuk-focus absolute left-4 top-4 disabled:pointer-events-none govuk-button govuk-button--secondary"
    type="button"
  >
    Close
  </button>
</div>
`;
